static/app/utils/replays/hooks/useReplayList.tsx

```diff
@@ -36,6 +36,42 @@ type State = {
 
 type Result = State;
 
+export async function fetchReplayList({
+  api,
+  organization,
+  location,
+  eventView,
+}): Promise<Result> {
+  try {
+    const path = `/organizations/${organization.slug}/replays/`;
+
+    const [{data: records}, _textStatus, resp] = await api.requestPromise(path, {
+      includeAllArgs: true,
+      query: {
+        ...eventView.getEventsAPIPayload(location),
+        cursor: location.query.cursor,
+      },
+    });
+
+    const pageLinks = resp?.getResponseHeader('Link') ?? '';
+
+    return {
+      fetchError: undefined,
+      isFetching: false,
+      pageLinks,
+      replays: records.map(mapResponseToReplayRecord),
+    };
+  } catch (error) {
+    Sentry.captureException(error);
+    return {
+      fetchError: error,
+      isFetching: false,
+      pageLinks: null,
+      replays: [],
+    };
+  }
+}
```

Comment:
refactor so I could call the function without the hooks wrapper stuff in the way.
Because there's a new hook with an async step in the middle, so this is the first way i sorted out how to solve for that and code share.